File: Data_Structure/C/BinarySearch.c
#include<stdio.h>

int BinarySearch(int arr[],int size, int target){
    int left = 0;
    int right = size-1;

    while(left <= right){
        int mid = left + (right-left)/2;

        if(arr[mid]==target){
            return mid;
        }else if(arr[mid]< target){
            left = mid + 1;
        }else{
            right = mid -1;
        }
    }
    return -1;
}

int main(){

    printf("enter the size of the array: ");
    int count;
    scanf("%d",&count);
    int array[count];
    printf("Enter the %d number of elements in sorted order\n",count);
    for(int i =0;i<count;i++){
        printf("Element %d : ",i+1);
        scanf("%d",&array[i]);
    }

    printf("Enter target to search:");
    int value;
    scanf("%d",&value);

    int index = BinarySearch(array,count,value);

    if(index == -1){
        printf("No element found");
    }else{
        printf("Element found at %d", index);
    }
    return 0;
}

File: Data_Structure/Java/BinarySearch.java

import java.util.Scanner;

public class BinarySearch {
    public static void main(String[] args){

        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the number of elements: ");
        int count = sc.nextInt();
        int [] arr = new int[count];

        System.err.println("Enter"+ count + "elemets in sorted order:\n");
        for(int i = 0; i< count; i++){
            System.err.println("Element "+i);
            arr[i] = sc.nextInt();
        }

        System.out.println("Enter the element to search: ");
        int value = sc.nextInt();
        int index = BSearch(arr, value);
        if(index != -1){
            System.out.println("Element found at index: " + index);
        }else{
            System.out.println("Element not found in the array.");
        }
    
    }

    public static int BSearch(int array[],int target){
        int left = 0;
        int right = array.length-1;
        while(left <= right){
            int mid = left + (right - left) / 2;

            if(array[mid] == target){
                return mid; // Target found
            }else if(array[mid] < target){
                left = mid + 1; // Search in the right half
            }else{
                right = mid - 1;
            }
        }
        return -1;
    }
    
}
